### PR TITLE
Expose Lens Monaco Editor component to extensions-api

### LINKED ISSUE #5094: Expose Lens Monaco Editor component to public
At present we are not able to use the Monaco Editor used by Lens. We were trying to use this since long time back but unfortunately with lens when we try to add this dependency and use, syntax highlighting was not working. So we have `js-yaml` 

W.r.to the latest lens CRD definitions, we see Monaco Editor used inside side panel. Can this be exposed for easy integration for us? This helps everyone.

### PR DESCRIPTION
**Pull request by ixrock**
Screenshot 2022-04-21 at 15 49 41

**Breaking change in extension's development process:**

- typescript version 4.5+ required to support type on import names, see: [typescript release notes](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names)

**How to fix:** _(to avoid compilation errors/warnings)_

`npm install typescript -D` or `yarn add typescript` _(latest version TS recommended)_

close #5094

### CODE REVIEW

**Review by aleksfront — COMMENTED** [review]
Got this errors trying to compile extension with new `k8slens/extensions` types:
Screenshot 2022-04-22 at 11 51 40

It points to the following line
new monaco line

**Comment by ixrock** [comment]
This happens because some extensions are using old typescript < 4.5 see: [typescript release notes](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names)
So, since it affects only development process, runtime should work as before. 
Moreover we agreed with @jakolehm that we should add our Lens release notes about it to help developers fix this issue faster.

**Review by aleksfront — DISMISSED** [review]
LGTM, works with fresh (at least 4.5.0) typescript version.

File: src/renderer/components/monaco-editor/monaco-editor.tsx [hunk 1]
```diff
@@ -55,22 +55,18 @@ export class MonacoEditor extends React.Component<MonacoEditorProps> {
   public staticId = `editor-id#${Math.round(1e7 * Math.random())}`;
   public dispose = disposer();
```
Comment (Nokel81) [inline src/renderer/components/monaco-editor/monaco-editor.tsx]: Shouldn't this also be private?

File: src/renderer/components/monaco-editor/monaco-editor.tsx [hunk 2]
```diff
@@ -55,22 +55,18 @@ export class MonacoEditor extends React.Component<MonacoEditorProps> {
...
 
-  // TODO: investigate how to replace with "common/logger"
-  //  currently leads for stucking UI forever & infinite loop.
-  //  e.g. happens on tab change/create, maybe some other cases too.
-  logger = console;
-
   @observable.ref containerElem: HTMLElement;
   @observable.ref editor: editor.IStandaloneCodeEditor;
```
Comment (Nokel81) [inline src/renderer/components/monaco-editor/monaco-editor.tsx:59]: Shouldn't these also be private?

**Review by Nokel81 — DISMISSED** [review]

File: src/renderer/components/monaco-editor/monaco-editor.tsx (hunk 2 above)
Comment (ixrock) [reply src/renderer/components/monaco-editor/monaco-editor.tsx:59]: Maybe `protected`, but not sure.. @jakolehm any suggestions?

File: src/renderer/components/monaco-editor/monaco-editor.tsx (hunk 1 above)
Comment (ixrock) [reply src/renderer/components/monaco-editor/monaco-editor.tsx]: Fixed

File: src/renderer/components/monaco-editor/monaco-editor.tsx (hunk 2 above)
Comment (ixrock) [reply src/renderer/components/monaco-editor/monaco-editor.tsx:59]: I think these ones should be public as if developer will decide to make a `ref={monaco => monaco.editor}` those would be available.

**Review by Nokel81 — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]

**Pull request merged**